services/QuillLMS/engines/comprehension/app/models/comprehension/automl_model.rb

```diff
@@ -115,7 +121,7 @@ def comprehension_name
     end
 
     private def automl_model_full_id
-      @model_full_id ||= automl_client.model_path(project: ENV['AUTOML_GOOGLE_PROJECT_ID'], location: ENV['AUTOML_GOOGLE_LOCATION'], model: automl_model_id)
+      @model_full_id ||= automl_client.model_path(project: ENV['AUTOML_GOOGLE_PROJECT_ID'], location: ENV['AUTOML_GOOGLE_LOCATION'], model: automl_model_id.strip)
```

Comment:
I might be confused here but it seems like this `strip` call is redundant given the `strip_whitespace` validation.

Reply:
Is this a convenience hack to avoid backfilling? I'm cool with that

Reply:
@brendanshean @happythenewsad In practice `automl_model_full_id` will almost always be called before `save`, and thus it will be called before `valid?`.  That means that the `before_validation` hooks won't have triggered before this function gets called in a number of scenarios.  So, yeah, kind of redundant, but I couldn't think of a cleaner way to handle this.

Reply:
Would `after_initialize` apply here?